pyqtorch/noise/readout.py

```diff
@@ -150,7 +157,13 @@ def create_confusion_matrices(noise_matrix: Tensor, error_probability: float) ->
     return torch.stack(confusion_matrices)
 
 
-class ReadoutNoise:
+class ReadoutInterface(ABC):
+    @singledispatchmethod
+    def apply(self, input_to_corrupt, n_shots):
```

Comment:
```suggestion
    def apply(self, input, n_shots):
```

But I suspect `input` is a reserved keyword.

Reply:
Maybe simply `inputs` ? Since in both cases you've got some iterable.